rustual-boy-libretro/src/lib.rs

```diff
@@ -60,6 +61,7 @@ pub struct Context {
     system: Option<System>,
     video_output_frame_buffer: OutputBuffer,
     audio_frame_buffer: Vec<AudioFrame>,
+    last_serialized_state: RefCell<Option<Vec<u8>>>,
```

Comment:
Do we actually want to use `RefCell` here? I'd argue that we should always have a mutable reference to a `Context` struct whenever we need to modify this. There are a couple places where our API impl uses `&self`, but I think those should be changed to `&mut self` to show that they can in fact change internal state (such as caching a value) instead of using a `Cell` for internal mutability. This would not only make the code easier to read but better communicate what's happening internally.